### DISCUSSION TITLE
RAW conversion fails when the container is running with a custom user ID

### DISCUSSION THREAD

**Discussion by Sanke123** [photoprism/photoprism#2256] [Q&A] [answered]
Hey there,
the conversion from RAW to JPEG is not working for me when uploading new pictures. Running with Docker on Unraid.

I'm getting this error:
```
2022-04-15 11:37:35 ERRO import: path lookup '/photoprism/.config/darktable' fails with: 'No such file or directory' in 2018/11/20181103_112229_D546B992.cr2 (convert to jpeg)
2022-04-15 11:37:35 INFO darktable-cli: converting 2018/11/20181103_112229_D546B992.cr2 to jpg
2022-04-15 11:37:35 DEBU failed renaming file, fallback to copy and delete: rename /photoprism/import/upload/1650015452001/8J0A0001.CR2 /photoprism/originals/2018/11/20181103_112229_D546B992.cr2: invalid cross-device link
2022-04-15 11:37:35 INFO import: moving main raw file 8J0A0001.CR2 to 2018/11/20181103_112229_D546B992.cr2
2022-04-15 11:37:35 INFO media: 8J0A0001.CR2 was taken at 2018-11-03 11:22:29 +0000 UTC (meta)
```

**Comment by lastzero** [comment, marked as answer]
See:
- https://github.com/photoprism/photoprism/issues/2262
- https://github.com/photoprism/photoprism/issues/2225

TL;DR: Fixed in our development preview build!

**Reply by lastzero** [reply]
Fixed with high probability... two problems:
- `PHOTOPRISM_DISABLE_CHOWN: "false"` disabled permission updates like any other non-empty value (so the solution is to remove this variable completely from `docker-compose.yml` for user ids other than 0 or 1000)
- even  custom cache/config dirs were specified for the `darktable-cli` command, it still tried to create directories in `$HOME` (the problem was fixed by setting `$HOME` to a "fake" home directory inside the storage folder)

GitHub Issue:
- https://github.com/photoprism/photoprism/issues/2262

**Comment by MEGAMENE** [comment]
Hi,

I'm currently having the same problem.
With the March 2 2022 build I had the "path lookup '/photoprism/.config/darktable' fails with: 'No such file or directory'" error.
After updating to the newest development preview build I now have the "path lookup '/.config/darktable' fails with: 'No such file or directory'" error.
I just can't get it to work with either version.

**Reply by lastzero** [reply]
What is the exact build number? What user is photoprism running as? Did you set any custom path environment variables, maybe through Portainer without being aware of it?

**Reply by MEGAMENE** [reply]
The build number is 220416-73be4df8-Linux-AMD64.
Photoprism is running as the default nobody user (99) of Unraid.
I don't think so? I'm using the Unraid Docker config that someone made, but looking at the settings I don't see any mappings other than the default ones.

Also I only get the error when importing new CR2 files.
When I run the index process I don't get any errors, but the pictures don't get converted either.

**Reply by lastzero** [reply]
Not an expert for the Unraid template, other defaults MIGHT fix it and also running the container as user ID 1000.

**Reply by lastzero** [reply]
Added an "unraid" user with id 99 to our preview image. Building right now, available soon. Let us know if this helps!
- see commit https://github.com/photoprism/photoprism/commit/b54cfeaf5fb55b48f39c9da6b1c96219b269be02

**Reply by MEGAMENE** [reply]
Thanks, this fixed it for me.
It's now working properly with the newest preview version.

**Comment by Ilovemyhous** [comment]
Hello!
I think I have the same problem, but with another error.
I use an Olympus E-M10MarkII, and PhotoPrism (`Build 220302-0059f429-Linux-AMD64`) is running from Openmediavault (6.0.22-1 (Shaitan)).

The error being; `import: FATAL ERROR: Creation of the user's processing profile directory "/photoprism/.config/RawTherapee/profiles" failed! in 2022/04/20220411_154840_5C95ACDB.orf (convert to jpeg) `

**Reply by lastzero** [reply]
Is this also happening with our preview build? 🤔

**Reply by Ilovemyhous** [reply]
I'm not sure if I can get the preview build.
I installed it trough the web dashboard.
I could _probably_ install trough the CLI, but I'm afraid to not choose the right disk/shared folder, and/or break the installed version.

**Reply by PitRejection2359** [reply]
I have this exact issue - I was about to start a new "ticket" (not sure if thats the right terminology - this is my first use of github). I'm running the docker on Unraid (as prepared by Ich777), and the error In get for all my DNGs is:

time="2022-04-22T22:31:27+01:00" level=warning msg="index: failed converting '_[filename]_.DNG' to jpeg (\nFATAL ERROR:\nCreation of the user's processing profile directory \"/photoprism/.config/RawTherapee/profiles\" failed!\n\n)"

Ich777 suggested deleting the docker and adding the variable HOME (as described [here](https://forums.unraid.net/topic/83786-support-ich777-application-dockers/?do=findComment&comment=1118607)), but that didn't work, with exactly the same result.

Should I raise a separate ticket?
